src/glb-director/glb_encap.c

```diff
@@ -131,19 +131,19 @@ static int extract_packet_fields_ipv4(glb_route_context *route_context)
 		}
 	} else {
 		const struct l4_ports_hdr *orig_l4_hdr = safely_get_next_header(route_context, struct l4_ports_hdr);
-		if (orig_l4_hdr == NULL) {
+		if (unlikely(orig_l4_hdr == NULL)) {
 			glb_log_info("lcore: -> unexpected: could not retrieve L4 header, but expected it to exist");
 			return -1;
 		}
 
 		// the simple case: pull all the fields
 		route_context->src_addr.ipv4 = ipv4_hdr->src_addr;
 		route_context->dst_addr.ipv4 = ipv4_hdr->dst_addr;
-		route_context->src_port = ntohs(orig_l4_hdr->src_port);
-		route_context->dst_port = ntohs(orig_l4_hdr->dst_port);
+		route_context->src_port = orig_l4_hdr->src_port;
+		route_context->dst_port = orig_l4_hdr->dst_port;
```

Comment:
These weren't yet used in https://github.com/github/glb-director/pull/39, but they should be stored/remain in network byte order to ensure the values going into the hash are consistent regardless of host byte order. For calculation purposes (like `flow_hash_hint`, host byte order was correct, so I've moved the network<->host byte order change to just that variable.